integration_tests/add-custom-barbell-exercise.integration.test.tsx

```diff
@@ -108,3 +116,107 @@ describe("User Journey: Add Custom Barbell Exercises", () => {
+      // Wait for a form draft to exist before re-rendering.
+      await waitFor(async () => {
+        const { data: drafts } = await supabase
+          .from("form_drafts")
+          .select("*")
+          .eq(
+            "user_id",
+            USER_ID["add-custom-barbell-exercise.integration.test.tsx"],
+          );
+        expect(drafts?.length).toBeGreaterThan(0);
+      });
+      page = await EquipmentExercisePage(pageProps);
+      await act(async () => render(page));
+    });
+
+    // Find the select reps component and click the AMRAP toggle
+    await act(async () => {
+      (
+        await Promise.all([
+          waitFor(() => screen.getByTestId(TestIds.SelectRepsAMRAPToggle)),
+          waitFor(() => screen.getByTestId(TestIds.SelectRepsAdd)),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(45))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(25))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(10))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(5))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(2.5))),
+          waitFor(() => screen.getByTestId(TestIds.PerceivedEffort("hard"))),
+          waitFor(() => screen.getByTestId(TestIds.WarmupToggle)),
+        ])
+      ).map((e) => e.click());
+      const notes = await waitFor(() => screen.getByTestId(TestIds.NotesInput));
+      fireEvent.change(notes, { target: { value: "Test note" } });
+    });
+
+    await act(async () => {
+      const addBarbellLiftButton = await waitFor(() =>
+        screen.getByTestId(TestIds.AddBarbellLiftButton),
+      );
+      addBarbellLiftButton.click();
+
+      // Wait for the new lift to exist in the DB before re-rendering.
+      await waitFor(async () => {
+        const { data: actualExercises } = await supabase
+          .from("exercises")
+          .select("*")
+          .eq(
+            "user_id",
+            USER_ID["add-custom-barbell-exercise.integration.test.tsx"],
+          )
+          .eq("exercise_type", "barbell_deadlift");
+        expect(actualExercises?.length).toBe(1);
+        const actualExercise = actualExercises![0];
+        const {
+          reps: actualReps,
+          is_amrap: actualIsAmrap,
+          actual_weight_value: actualActualWeightValue,
+          target_weight_value: actualTargetWeightValue,
+          perceived_effort: actualPerceivedEffort,
+          warmup: actualIsWarmup,
+          notes: actualNotes,
+        } = actualExercise;
+        expect(actualReps).toBe(6);
+        expect(actualIsAmrap).toBe(true);
+        expect(actualActualWeightValue).toBe(220);
+        expect(actualTargetWeightValue).toBe(220);
+        expect(actualPerceivedEffort).toBe("hard");
+        expect(actualIsWarmup).toBe(true);
+        expect(actualNotes).toBe("Test note");
+      });
+    });
+  });
+});
+
+// Page props do not differ for any of the integration tests.
+const pageProps: EquipmentExercisePageProps = {
```

Comment:
I just refactored this to the bottom "out of the way" because it's shared for all the integration tests and it makes it where you don't have to look through as much code to actually get to the tests.